src/core/reactor_backend.cc

```diff
@@ -44,6 +46,11 @@ namespace fs = std::filesystem;
 class pollable_fd_state_completion : public kernel_completion {
     promise<> _pr;
 public:
+    void abort() {
```

Comment:
please mark as `noexcept`

I suggest to add a `std::optional<std::exception_ptr>` parameter to the abort interface so a different exception can be set by the caller.  E.g.
`_pr.set_exception(opt_ex.value_or(std::make_exception_ptr(pollable_fd_aborted())))`

Please see https://github.com/scylladb/seastar/pull/1088 for example.

Reply:
I read #1088, fixed code as you suggested and also added virtual function to override default exception. Perhaps, last part is a bit over-engineered.